### PR TITLE
Undraft the bylaws: IP Policy [due Oct 27]

### PR DESCRIPTION
**Pull request by ebullient**
[🗳️ Vote progress](https://github.com/commonhaus/foundation/pull/178#issuecomment-2240642060 "IC_kwDOKRPTI86FjXwM")

Continuing with our reviews: the Intellectual Property policy.

voting group: @commonhaus/cf-egc 

Do one of the following:

- Approve the PR or react with 👍 (:+1:) if it looks good to you
- Review with Comments or react with 👀 (:eyes:) if you're "ok" with it (it may not be your favorite)
- If you think it needs discussion or revision
    - Create a review, add your comments and require changes
    - Use the +- button to make a suggestion (instead of just adding a comment).

### CODE REVIEW

File: policies/ip-policy.md [hunk 1]
```diff
@@ -33,21 +30,19 @@ The Project Documentation License can be any OSI-approved license or a Creative
 
 ## Contributions
 
-> [!TIP]
-> All code contributions must align with the project’s license and come with a promise (DCO) from contributors that they have the right to contribute under these terms.
-
-All new code contributions to any Project shall be made under the Project Code License accompanied by a Developers Certificate of Origin (DCO, available at <http://developercertificate.org/>), which will bind the individual contributor and, if applicable, their employer to the Project Code License.
+All new code contributions to any Project must be made under the Project Code License. Contributions must be accompanied by either a Developer Certificate of Origin (DCO available at <http://developercertificate.org/>) or a Contributor License Agreement (CLA), as determined by the project’s technical governing body. The DCO or CLA binds the individual contributor and, if applicable, their employer to the Project Code License.
```
Comment (maxandersen) [inline policies/ip-policy.md:33]: I'm assuming it is implied that all projects have either a dco or cla in effect for any new contributors?

If it is - should it not be explicitly stated as reading this a project don't need one - just that if it has either dco or cla those are in effect.

Comment (ebullient) [reply policies/ip-policy.md:33]: I am not sure what is unclear.

All new code contributions to a CF project must follow the project code license and be accompanied by either the DCO or a CLA (as chosen by the project). I don't think there is anything in that text that says "neither" or "none" is a valid option.

**Review by aalmiray — APPROVED** [review]

**Review by cealsair — APPROVED** [review]

**Comment by jkschneider** [comment]
> Incorrect <CF Mark> by <your company name>

OpenRewrite objects to this constraint. I think it is incompatible with CommonHaus’ general policy that the foundation doesn’t interefere in the governance of projects. It is clear that Moderne is the governing maintainer of OpenRewrite (at the current time) regardless of who legally owns the trademark.

**Comment by ebullient** [comment]
extending deadline for detailed review

**Review by velo — APPROVED** [review]

**Review by evanchooly — APPROVED** [review]

**Review by JooHyukKim — APPROVED** [review]

**Comment by ebullient** [comment]
Setting due date to end of October.

**Comment by ebullient** [comment]
Final draft. A sanity check would be lovely.

File: policies/ip-policy.md [hunk 2]
```diff
@@ -28,54 +24,51 @@ The CF is most familiar with the following four licenses, which can simplify the
...
-The technical governing body of each Project may provide for additional requirements with respect to contributions. For example, a Project may require that new code contributions to a Project be accompanied by a signed Contributor License Agreement (CLA).
+The DCO or CLA binds the individual contributor and, if applicable, their employer to the Project Code License.
 
 ### Developers Certificate of Origin (DCO) or Contributor License Agreement (CLA)?
 
 The DCO and CLA differ in their approach to intellectual property rights.
 
-- A DCO is a straightforward way for contributors to assert their rights to submit work, typically through a declaration in the commit message.
-- A CLA is a formal agreement that allows contributors to grant broader rights over their contributions to the project.
+- A DCO is a straightforward method for contributors to confirm they have the necessary rights to submit work, typically through a declaration in the commit message.
```
Comment (gavinking) [inline policies/ip-policy.md:42]:
```suggestion
- A DCO is a straightforward method for contributors to declare that they have the necessary rights to submit work, typically through a declaration in the commit message.
```

Comment (ebullient) [reply policies/ip-policy.md:42]: `confirm` was used on recommendation from our lawyers. So while I agree that affirm or declare would make sense, I'll follow guidance in this case.

**Comment by ebullient** [comment]
vote::result No voiced objections with ample bake time; proceeding.

**Pull request merged**